### PR TITLE
ERSSUP-57362- OAS: Add SPCA role to SPC-authorised endpoints and add OBO User ID header

### PR DESCRIPTION
**Pull request by nhsd-nigel-mellor**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/components/schemas/headers/request/OboUserId.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+in: header
+name: nhsd-ers-obo-user-id
+description: | 
+  The (SDS) user ID of the user that the authenticating user wishes to act on behalf of (OBO). Currently "on behalf of" is only supported for a Service Provider Clinician Admin (SPCA) acting on behalf of a Service Provider Clinician (SPC). Where an OBO User ID is supplied then the authenticating user should therefore be an SPCA and the OBO User ID should be the ID of an appropriate SPC.
```
Comment (jamiegreen4) [inline specification/components/schemas/headers/request/OboUserId.yaml:4]: If it hasn't already been reviewed, it's very likely that NHSD will want to review and sign-off this wording. OdsCodeOnlyUserRestricted.yaml when through a similar review cycle

File: specification/e-referrals-service-api.yaml [hunk 2]
```diff
@@ -489,6 +489,9 @@ paths:
           - `REFERRING_CLINICIAN`
           - `REFERRING_CLINICIAN_ADMIN`
           - `SERVICE_PROVIDER_CLINICIAN`
+          - `SERVICE_PROVIDER_CLINICIAN_ADMIN`
+
+        If the `SERVICE_PROVIDER_CLINICIAN_ADMIN` role is used then the User ID of Service Provider Clinician that the Admin is acting on behalf of must be supplied via the `nhsd-ers-obo-user-id` header in the request.
```
Comment (jamiegreen4) [inline specification/e-referrals-service-api.yaml:494]: Should this start "If the `SERVICE_PROVIDER_CLINICIAN_ADMIN` role is used then the User ID of the Service..."?

**Review by nhsd-jack-wainwright — APPROVED** [review]

**Pull request merged**